crates/base-db/src/input.rs

```diff
@@ -296,6 +296,7 @@ pub struct CrateData {
     pub dependencies: Vec<Dependency>,
     pub origin: CrateOrigin,
     pub is_proc_macro: bool,
+    pub cwd: Option<AbsPathBuf>,
```

Comment:
```suggestion
    /// The working directory to run proc-macros in. This is the workspace root of the cargo workspace for workspace members, the crate manifest dir otherwise.
    pub proc_macro_cwd: Option<AbsPathBuf>,
```